src/storage/service.ts

```diff
@@ -924,4 +925,23 @@ export class StorageService {
   async download (commp: string | CommP, options?: DownloadOptions): Promise<Uint8Array> {
     return await this.providerDownload(commp, options)
   }
+
+  /**
+   * Get proof set roots for this storage service's proof set
+   * @returns Array of root CIDs as CommP objects
+   */
+  async getProofSetRoots (): Promise<CommP[]> {
+    const proofSetData = await this._pdpServer.getProofSet(this._proofSetId)
+    return proofSetData.roots.map(root => {
+      const commP = asCommP(root.rootCid)
+      if (commP == null) {
+        throw createError(
+          'StorageService',
+          'getProofSetRoots',
+          `Invalid CommP received from server: ${root.rootCid.toString()}`
+        )
+      }
```

Comment:
don't we know that it's a CommP already from our `asProofSet` conversion? double-check that, maybe all you need to do is `map(root => root.rootCid)` and you're good to go?